ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/TransactionState.kt

```diff
@@ -0,0 +1,25 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.ledger.common.Party
+
+/**
+ * Defines a transaction state, composed of a [ContractState] and associated transaction state information.
+ *
+ * @param T The underlying type of the [ContractState] instance.
+ * @property contractState The [ContractState] of the current [TransactionState] instance.
+ * @property contractStateType The [ContractState] type of the current [TransactionState] instance.
+ * @property contractType The [Contract] type of the current [TransactionState] instance.
+ * @property notary The notary of the current [TransactionState] instance.
+ * @property encumbrance The encumbrance of the current [TransactionState] instance.
+ */
+@DoNotImplement
+@CordaSerializable
+interface TransactionState<out T : ContractState> {
+    val contractState: T
+    val contractStateType: Class<out T>
```

Comment:
What do we think `contractStateType` would be used for?

Reply:
@mnesbit You mentioned that we needed to know the specific type; presumably because of type information loss at serialization, and we need to cast it back?

Reply:
To prove that you have revealed all cash output states to an oracle for instance. The more important use for this data is in the under the hood info component group, which we have to use to handle validation and deserialization